src/eth/executor.rs

```diff
@@ -21,29 +24,42 @@ use crate::eth::primitives::StoragePointInTime;
 use crate::eth::primitives::TransactionExecution;
 use crate::eth::primitives::TransactionInput;
 use crate::eth::storage::EthStorage;
+use crate::ext::new_tokio_runtime;
+
+/// Number of events in the backlog.
+const NOTIFIER_CAPACITY: usize = u16::MAX as usize;
+
+type EvmTask = (EvmInput, oneshot::Sender<anyhow::Result<TransactionExecution>>);
 
 /// The EthExecutor struct is responsible for orchestrating the execution of Ethereum transactions.
 /// It holds references to the EVM, block miner, and storage, managing the overall process of
 /// transaction execution, block production, and state management.
 pub struct EthExecutor {
-    // Mutex-wrapped EVM for synchronized access to EVM operations.
-    evm: Mutex<Box<dyn Evm>>,
+    // Tokio runtime where background EVMs are executed (keep reference because it cannot be dropped).
+    _runtime_guard: Runtime,
```

Comment:
In the comment abote you say we should keep a reference to Runtime because it cannot be dropped. But you're keeping the value, is this intentional? Did I misunderstand?